Adaptors/MongoDB/src/Common/MongoCollectionProvider.cs

```diff
@@ -182,19 +195,49 @@ await model.ShardCollectionAsync(session,
         catch (Exception ex)
         {
           lastException = ex;
-          logger.LogDebug(ex,
-                          "Retrying to shard {CollectionName} collection",
-                          model.CollectionName);
+          logger.LogWarning(ex,
+                            "Retrying to shard {CollectionName} collection",
+                            model.CollectionName);
           await Task.Delay(1000 * indexRetry,
                            cancellationToken)
                     .ConfigureAwait(false);
         }
       }
     }
 
+    for (var indexRetry = 1; indexRetry < options.MaxRetries; indexRetry++)
+    {
+      lastException = null;
+      try
+      {
+        await model.InitializeCollectionAsync(session,
+                                              output,
+                                              initDatabase)
+                   .ConfigureAwait(false);
+        break;
+      }
+      catch (MongoBulkWriteException<TData> e) when (e.WriteErrors.All(error => error.Category == ServerErrorCategory.DuplicateKey))
+      {
+        logger.LogWarning(e,
+                          "Values were already present within the collection {CollectionName}",
+                          model.CollectionName);
```

Comment:
I am not sure Warning is the good level for this. You would get it every time your pod start.